ftpvl/processors.py

```diff
@@ -438,3 +440,105 @@ def process(self, b: Evaluation) -> Evaluation:
         difference_eval = Evaluation(diff)
 
         return difference_eval
+
+class FilterByIndex(Processor):
+    """
+    Processor that filters an Evaluation by comparing a specified value
+    after indexing.
+
+    This is best used in a processing pipeline after the Reindex processor.
+    For filtering an evaluation based on metrics (which is not an index),
+    use the FilterByMetric processor.
+
+    Parameters
+    ----------
+    index_name : str
+        the name of the index to use when filtering
+    index_value : Any
+        the value to compare with
+    
+    Examples
+    --------
+    >>> a = Evaluation(pd.DataFrame(
+    ... data=[
+    ...     {"x": 1, "y": 5},
+    ...     {"x": 4, "y": 10}
+    ... ],
+    ... index=pd.Index(["a", "b"], name="key")))
+    >>> a.process([FilterByIndex("key", "a")]).get_df()
+        x    y
+    key
+    a   1    5
+    """
+    def __init__(self, index_name: str, index_value: Any):
+        self.index_name = index_name
+        self.index_value = index_value
+    
+    def process(self, input_eval: Evaluation):
+        old_df = input_eval.get_df()
+        if isinstance(old_df.index, pd.MultiIndex):
+            new_df = old_df.xs(self.index_value, level=self.index_name)
+        elif isinstance(old_df.index, pd.Index):
+            # slicing instead of indexing to maintain shape
+            new_df = old_df.loc[self.index_value:self.index_value]
+        else:
+            raise ValueError("Incompatible dataframe index.")
+        return Evaluation(new_df, input_eval.get_eval_id())
+
+class Aggregate(Processor):
+    """
+    Processor that allows you to aggregate an entire Evaluation using a specified
+    function.
+
+    This is useful for custom aggregators and as a superclass for specific
+    aggregator implementations, such as GeomeanAggregate.
```

Comment:
I see. Its init function takes an extra aggregation function argument, and subclasses supply a specific aggregation function. Perhaps this may be better phrased as follows:
```suggestion
    This acts as a superclass for specific aggregator implementations, such as GeomeanAggregate.
    It can also be used for custom aggregations, by supplying an aggregator function to the constructor.
```